front/app/containers/InitiativesShow/Cosponsors/ListOfCosponsors.tsx

```diff
@@ -1,25 +1,67 @@
 import React from 'react';
-import { Box } from '@citizenlab/cl2-component-library';
+import { Box, Text } from '@citizenlab/cl2-component-library';
 import Avatar from 'components/Avatar';
 import { IInitiativeCosponsorship } from 'api/initiatives/types';
 import UserName from 'components/UI/UserName';
+import { useIntl } from 'utils/cl-intl';
+import messages from '../messages';
 
 interface Props {
   cosponsorships: IInitiativeCosponsorship[];
 }
 
+const AcceptedCosponsor = ({
+  cosponsorship,
+}: {
+  cosponsorship: IInitiativeCosponsorship;
+}) => {
+  return (
+    <Box display="flex" alignItems="center">
+      <Box mr="4px">
+        <Avatar userId={cosponsorship.user_id} size={32} />
+      </Box>
+      <Box mr="4px">
+        <UserName userId={cosponsorship.user_id} isLinkToProfile />
+      </Box>
+    </Box>
+  );
+};
+
+const PendingCosponsor = ({
+  cosponsorship,
+}: {
+  cosponsorship: IInitiativeCosponsorship;
+}) => {
+  const { formatMessage } = useIntl();
+
+  return (
+    <Box display="flex" alignItems="center">
+      <Box mr="4px">
+        <Avatar userId={cosponsorship.user_id} size={32} />
+      </Box>
+      <Box mr="4px">
+        <UserName userId={cosponsorship.user_id} isLinkToProfile italic />
+      </Box>
+      <Text fontStyle="italic">({formatMessage(messages.pending)})</Text>
+    </Box>
+  );
+};
+
 const ListOfCosponsors = ({ cosponsorships }: Props) => {
+  const acceptedCosponsorships = cosponsorships.filter(
+    (c) => c.status === 'accepted'
```

Comment:
It would be so much nicer if we could do sth like `initiative.data.attributes.cosponsorships.accepted()` in the object-oriented style. Is it possible?

Currently, we have `c.status === 'accepted'` repeated 6 times.

Reply:
Not possible to my knowledge, except if we build/use a util.